lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs

```diff
@@ -168,11 +170,54 @@ simpleSpec = do
             runExceptT (putCheckpoint db testPk testCp) `shouldReturn` Right ()
             readCheckpoint db testPk `shouldReturn` Just testCp
 
+loggingSpec :: Spec
+loggingSpec = withLoggingDB $ do
+    describe "Sqlite query logging" $ do
+        it "should log queries at DEBUG level" $ \(db, getLogs) -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            logs <- logMessages <$> getLogs
+            logs `shouldHaveLog` (Debug, "INSERT")
+
+        it "should not log query parameters" $ \(db, getLogs) -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            let walletName = T.unpack $ coerce $ name testMetadata
+            msgs <- T.unlines . map snd . logMessages <$> getLogs
+            T.unpack msgs `shouldNotContain` walletName
+
+    describe "Sqlite observables" $ do
+        it "should measure query timings" $ \(db, getLogs) -> do
+            replicateM_ 5 $ listWallets db
+            msgs <- T.unlines . map snd . logMessages <$> getLogs
+            T.unpack msgs `shouldContain` "runQuery monitor works"
```

Comment:
I see failure of this in CI : 
```
    Failures:
    
      test/unit/Cardano/Wallet/DB/SqliteSpec.hs:191:13: 
      1) Cardano.Wallet.DB.Sqlite, Sqlite observables, should measure query timings
           "SELECT \"wallet_id\" FROM \"wallet\" ORDER BY \"wallet_id\"\nSELECT \"wallet_id\" FROM \"wallet\" ORDER BY \"wallet_id\"\nSELECT \"wallet_id\" FROM \"wallet\" ORDER BY \"wallet_id\"\nSELECT \"wallet_id\" FROM \"wallet\" ORDER BY \"wallet_id\"\nSELECT \"wallet_id\" FROM \"wallet\" ORDER BY \"wallet_id\"\n" does not contain "runQuery monitor works"
```
looks like `newMemoryDBLayer` has not taken **[CreateMessage Info "runQuery monitor works"]** into account